src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -423,7 +455,7 @@ void startProcessing() {
             public void run() {
                 LOGGER.info("Started offer processing thread: " + threadName);
                 try {
-                    while (true) {
+                    while (isRunning()) {
```

Comment:
What's the guarantee that between the `isRunning()` check and `processOffers()` below that the `driver` wont be set to null?